taxi/views.py

```diff
@@ -1,12 +1,19 @@
+from django.contrib.auth.mixins import LoginRequiredMixin
 from django.shortcuts import render
 from django.views import generic
+from django.contrib.auth.decorators import login_required
 
 from .models import Driver, Car, Manufacturer
 
 
+@login_required
 def index(request):
     """View function for the home page of the site."""
 
+    num_visits = request.session.get("num_visits", 0)
```

Comment:
The code for handling 'num_visits' should ensure that it works as expected when visiting the page for the first time. According to the checklist, when you visit the page for the first time, there should be: 'You have visited this page 1 time'. Make sure this logic is correctly implemented.